Add room management logic to WebSocket server

Otaku17 opened the pull request:
Introduces a RoomManager class for handling room creation, joining, and leaving. Integrates room-related events ('create_room', 'join_room', 'leave_room') into the WebSocket server, enabling clients to manage multiplayer rooms and receive updates.

Otaku17 committed:
Add room management logic to WebSocket server

Introduces a RoomManager class for handling room creation, joining, and leaving. Integrates room-related events ('create_room', 'join_room', 'leave_room') into the WebSocket server, enabling clients to manage multiplayer rooms and receive updates.

Menthx commented:
Il faudrait pointer sur develop ducoup.

AerunDev commented on src/websocket/logic/server.ts:
```diff
@@ -291,7 +329,7 @@ export class Server {
         console.error(`Error handling event ${event}:`, error);
       }
-    } else {
+    } else if (!['create_room', 'join_room', 'leave_room'].includes(event)) {
```
On peut peut-être utiliser une variable ou une constante ici pour faciliter la gestion des différents événements ?

Menthx replied:
En effet, je me suis permis de variabiliser ce qui pouvais l'être !

Menthx committed:
feat: Add room and error handling types to WebSocket server

Introduces new types and constants for room management and error codes. Adds a Room interface and RoomId type for better structure in room handling. Implements ROOM_EVENTS and ERROR_CODES for improved event management and error reporting in the WebSocket server.

Menthx committed:
Merge branch 'dev' of https://github.com/PokemonWorkshop/OnlineServer into Adding-room-system

Menthx commented:
Add a GetRoom list and get by id for later.

Menthx committed:
feat: Enhance room management with new event handling

Adds 'get_room_by_id' and 'get_all_rooms' events to ROOM_EVENTS and ROOM_OUTBOUND_EVENTS for improved room retrieval functionality. Updates RoomManager methods to public access and implements corresponding logic in the WebSocket server for handling these new events.
